modules/portal/portal-cluster/src/main/java/com/liferay/portal/cluster/internal/ClusterMasterExecutorImpl.java

```diff
@@ -185,58 +178,25 @@ protected void deactivate() {
 		_localClusterNodeId = null;
 	}
 
-	protected String getMasterClusterNodeId(boolean notify) {
-		String owner = null;
+	protected String getMasterClusterNodeId(Address address) {
+		if (_master) {
+			return _localClusterNodeId;
+		}
 
 		while (true) {
```

Comment:
This while loop can not be right.
Let's say right after you got the current parameter address, that node died, which means the following _clusterExecutorImpl.getClusterNode(address) will always return null, then it gets into an infinite loop with info log.